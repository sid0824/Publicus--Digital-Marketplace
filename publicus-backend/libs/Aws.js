import AWS from 'aws-sdk';
import {AccessKeyId, SecretAccessKey} from "../config/ConfigValues";

export const s3 = new AWS.S3({
  region: 'ap-southeast-2',
  accessKeyId: AccessKeyId,
  secretAccessKey: SecretAccessKey
});

export const ses = new AWS.SES({
  region: 'us-west-2'
});