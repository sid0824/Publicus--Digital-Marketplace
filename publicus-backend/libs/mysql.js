import mysql from 'mysql';
import {MySqlHost, MySqlPassword, MySqlUser} from "../config/ConfigValues";

//mysql connection to database
const connection = mysql.createConnection({
    host: MySqlHost,
    user: MySqlUser,
    password: MySqlPassword,
    database: "survey",
    multipleStatements: true
});

export const runQuery = async (query) => {
    return new Promise((resolve, reject) => {
        connection.query(query, (error, results) => {
            if(error) {
                reject(error);
            } else {
                resolve(results);
            }
        });
    });
};