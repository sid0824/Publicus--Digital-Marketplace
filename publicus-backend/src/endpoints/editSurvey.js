import {failure, success} from "../util/ResponseUtil";
import {runQuery} from "../../libs/mysql";
import {deleteTagsBySurvey, insertTagsQuery} from "../util/QueryUtil";

// Update the existing form
const updateSurveyQuery = (surveyId, name, price, form, description, redirect, instructions) => {
  return `UPDATE Surveys 
          SET name = '${escape(name)}', price = ${price}, form = '${escape(JSON.stringify(form))}', 
           description = '${escape(description)}', redirect = '${escape(redirect)}', instructions = '${escape(instructions)}'
          WHERE surveyId = '${surveyId}'`;
};

export async function main(event, context) {
  try {
    const data = JSON.parse(event.body);

    // update form values
    await runQuery(updateSurveyQuery(data.surveyId, data.name, data.price, data.form, data.description, data.redirect, data.instructions));

    // delete existing tags
    await runQuery(deleteTagsBySurvey(data.surveyId));

    // insert new tags
    if (data.tags && data.tags.length) {
      await runQuery(insertTagsQuery(data.surveyId, data.tags));
    }
    return success();
  } catch (err) {
    console.error(err);
    return failure();
  }
}