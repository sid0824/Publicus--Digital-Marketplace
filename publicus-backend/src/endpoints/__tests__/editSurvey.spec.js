import { main } from '../editSurvey';
import { runQuery } from "../../../libs/mysql";

jest.mock("../../../libs/mysql");

const mockEvent = {
  requestContext: {
    identity: {
      cognitoIdentityId: 'MOCK_COGNITO_ID'
    },
  },
  body: JSON.stringify({
    surveyId: 'SURVEY_ID',
    name: 'NAME',
    price: 5,
    form: [{ MOCK: 'FORM' }],
    description: 'DESCRIPTION',
    redirect: 'REDIRECT',
    instructions: 'INSTRUCTIONS',
    tags: ["TAG1", "TAG2"]
  }),
  pathParameters: {
    surveyId: 'MOCK_SURVEY_ID'
  }
};

describe('editSurvey', () => {

  beforeEach(() => {
    runQuery.mockResolvedValue(null)
  });

  afterEach(() => {
    runQuery.mockReset();
  });

  test(`should respond with failure when an error occurs`, async () => {
    runQuery.mockReset();
    runQuery.mockImplementation(() => {
      throw new Error('ERROR')
    });
    const response = await main(mockEvent);
    expect(response).toMatchSnapshot();
  });

  test(`should edit an existing survey`, async () => {
    const response = await main(mockEvent);
    expect(runQuery.mock.calls[0][0]).toMatchSnapshot();
    expect(response).toMatchSnapshot();
  })

});