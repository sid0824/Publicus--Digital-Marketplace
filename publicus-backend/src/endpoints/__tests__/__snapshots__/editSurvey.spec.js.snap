// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`editSurvey should edit an existing survey 1`] = `
"UPDATE Surveys 
          SET name = 'NAME', price = 5, form = '%5B%7B%22MOCK%22%3A%22FORM%22%7D%5D', 
           description = 'DESCRIPTION', redirect = 'REDIRECT', instructions = 'INSTRUCTIONS'
          WHERE surveyId = 'SURVEY_ID'"
`;

exports[`editSurvey should edit an existing survey 2`] = `
Object {
  "body": undefined,
  "headers": Object {
    "Access-Control-Allow-Credentials": true,
    "Access-Control-Allow-Origin": "*",
  },
  "statusCode": 200,
}
`;

exports[`editSurvey should respond with failure when an error occurs 1`] = `
Object {
  "body": undefined,
  "headers": Object {
    "Access-Control-Allow-Credentials": true,
    "Access-Control-Allow-Origin": "*",
  },
  "statusCode": 500,
}
`;
